tscreen_posix.go

```diff
@@ -17,150 +17,65 @@
 package tcell
 
 import (
+	"fmt"
+	"github.com/pkg/term/termios"
+	"golang.org/x/sys/unix"
 	"os"
 	"os/signal"
 	"syscall"
 )
 
-// #include <termios.h>
-// #include <sys/ioctl.h>
-//
-// int getwinsize(int fd, int *cols, int *rows) {
-// #if defined TIOCGWINSZ
-//	struct winsize w;
-//	if (ioctl(fd, TIOCGWINSZ, &w) < 0) {
-//		return (-1);
-//	}
-//	*cols = w.ws_col;
-//	*rows = w.ws_row;
-//	return (0);
-// #else
-//	return (-1);
-// #endif
-// }
-//
-// int getbaud(struct termios *tios) {
-//     switch (cfgetospeed(tios)) {
-// #ifdef B0
-//     case B0: return (0);
-// #endif
-// #ifdef B50
-//     case B50: return (50);
-// #endif
-// #ifdef B75
-//     case B75: return (75);
-// #endif
-// #ifdef B110
-//	case B110: return (110);
-// #endif
-// #ifdef B134
-//	case B134: return (134);
-// #endif
-// #ifdef B150
-//	case B150: return (150);
-// #endif
-// #ifdef B200
-//	case B200: return (200);
-// #endif
-// #ifdef B300
-//	case B300: return (300);
-// #endif
-// #ifdef B600
-//	case B600: return (600);
-// #endif
-// #ifdef B1200
-//	case B1200: return (1200);
-// #endif
-// #ifdef B1800
-//	case B1800: return (1800);
-// #endif
-// #ifdef B2400
-//	case B2400: return (2400);
-// #endif
-// #ifdef B4800
-//	case B4800: return (4800);
-// #endif
-// #ifdef B9600
-//	case B9600: return (9600);
-// #endif
-// #ifdef B19200
-//	case B19200: return (19200);
-// #endif
-// #ifdef B38400
-//	case B38400: return (38400);
-// #endif
-// #ifdef B57600
-//	case B57600: return (57600);
-// #endif
-// #ifdef B76800
-//	case B76800: return (76800);
-// #endif
-// #ifdef B115200
-//	case B115200: return (115200);
-// #endif
-// #ifdef B153600
-//	case B153600: return (153600);
-// #endif
-// #ifdef B230400
-//	case B230400: return (230400);
-// #endif
-// #ifdef B307200
-//	case B307200: return (307200);
-// #endif
-// #ifdef B460800
-//	case B460800: return (460800);
-// #endif
-// #ifdef B921600
-//	case B921600: return (921600);
-// #endif
-//	}
-//	return (0);
-// }
-import "C"
-
-type termiosPrivate struct {
-	tios C.struct_termios
-}
+type termiosPrivate syscall.Termios
+
+var tiosp = termiosPrivate{}
 
-func (t *tScreen) termioInit() error {
-	var e error
-	var rv C.int
-	var newtios C.struct_termios
-	var fd C.int
+func (t *tScreen) termioInit() (err error) {
+	private := syscall.Termios{}
 
-	if t.in, e = os.OpenFile("/dev/tty", os.O_RDONLY, 0); e != nil {
-		goto failed
+	if t.in, err = os.OpenFile("/dev/tty", os.O_RDONLY, 0); err != nil {
+		return err
 	}
-	if t.out, e = os.OpenFile("/dev/tty", os.O_WRONLY, 0); e != nil {
-		goto failed
+	if t.out, err = os.OpenFile("/dev/tty", os.O_WRONLY, 0); err != nil {
+		return err
```

Comment:
This should probably close t.in...

Reply:
This happens in the deffered function 5 lines bleow this comment :)

Reply:
Right.  But that deferred thing won't get executed, because the defer needs to be moved ahead of this.